File: Driver/Safari/Selenium_assignment_safari/src/safari_driver_assignment_files/upload_file.java
package safari_driver_assignment_files;

import org.openqa.selenium.By;
import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.remote.LocalFileDetector;
import org.openqa.selenium.remote.RemoteWebDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class upload_file {

  public static final String AUTOMATE_USERNAME = "pratikpathare_oMjcIz";
  public static final String AUTOMATE_KEY = "EeUYtZMrb6bLzmGQoszr";
  public static final String URL = "https://" + AUTOMATE_USERNAME + ":" + AUTOMATE_KEY + "@hub-cloud.browserstack.com/wd/hub";

  public static void main(String[] args) throws Exception {

    DesiredCapabilities caps = new DesiredCapabilities();
    
    caps.setCapability("os", "OS X");
    caps.setCapability("os_version", "Monterey");
    caps.setCapability("browser", "Safari");
    caps.setCapability("browser_version", "15.0");
    caps.setCapability("browserstack.local", "false");
    caps.setCapability("browserstack.selenium_version", "3.14.0");

    caps.setCapability("name", "PHP_Java_File_Upload");

    RemoteWebDriver driver = new RemoteWebDriver(new java.net.URL(URL), caps);
    driver.setFileDetector(new LocalFileDetector());
    driver.get("http://www.fileconvoy.com/");
    driver.findElement(By.id("upfile_0")).sendKeys("/Users/pratikhemantpathare/Desktop/Demo.txt");
    driver.findElement(By.name("upload_button")).submit();
    JavascriptExecutor jse = (JavascriptExecutor)driver;
    System.out.println("Upload OK");
    driver.quit();
  }
}

File: Grid/Standalone/Selenium_with_Node_js/upload_file.js
const webdriver = require("selenium-webdriver");
const remote = require("selenium-webdriver/remote");

// Input capabilities
const capabilities = {
  "browserName": "Internet Explorer",
  "browser_version": "11.0",
  "os": "Windows",
  "os_version": "10",
  "name": "Bstack-[NodeJS] Upload Test",
  "browserstack.sendKeys": "true",
  "browserstack.debug": "true",
  "browserstack.user": "pratikpathare_oMjcIz",
  "browserstack.key": "EeUYtZMrb6bLzmGQoszr"
};

const driver = new webdriver.Builder()
  .usingServer("https://hub-cloud.browserstack.com/wd/hub")
  .withCapabilities(capabilities)
  .build();

//This will detect your local file
driver.setFileDetector(new remote.FileDetector());

(async () => {
  await driver.get("http://www.fileconvoy.com");

  const filePathElement = await driver.findElement(webdriver.By.id("upfile_0"));
  await filePathElement.sendKeys("/Users/pratikhemantpathare/Desktop/Demo.txt");

  await (await driver.findElement(webdriver.By.id("readTermsOfUse"))).click();
  await (await driver.findElement(webdriver.By.name("upload_button"))).click();
  
  await driver.quit();
})();
